docs/chipflow-commands.rst

```diff
@@ -0,0 +1,46 @@
+The ``chipflow`` command
+==========================
+
+The ``chipflow`` tool enables you to simulate your design or submit it to the ChipFlow backend.
+It implements several subcommands, which can be customised or added to in the ``steps`` section of :ref:`chipflow.toml<chipflow-toml-steps>`.
+
+.. _chipflow_toml: chipflow-toml-guide.rst
+
+``chipflow pin lock``
+---------------------
+
+The ``chipflow pin lock`` command performs pin locking for the current design.
+For every new top level interface with containing external pins with a ``PinSignature`` that is discovered, the necessary number of package pins is allocated and the mapping saved in the ``pins.lock`` file.
+This means that, unless the ``pins.lock`` file is deleted or manually modified, the pin assignments of all existing pins will always remain the same.
+
+``chipflow silicon``
+--------------------
+
+The ``chipflow silicon`` subcommand is used to send the design to the backend for build and tapeout. In general, it would be run inside GitHub Actions as part of a CI job.
+
+ - ``chipflow silicon prepare`` links the design, including all Amaranth modules and any external Verilog components, into a single RTLIL file that is ready to be sent to the backend.
+ - ``chipflow silicon submit`` sends the linked design along with the ``pins.lock`` file containing pinout information to the ChipFlow backend for the build. With the ``--dry-run`` argument, it can be used for a local test that the design is ready to be submitted
+
+Submitting the design to the backend requires the ``CHIPFLOW_API_KEY`` environment variable to be set with a valid API key obtained from the cloud interface.
+
+``chipflow sim``
+----------------
+
+The ``chipflow sim build`` command is used to build a CXXRTL simulation of the design; converting the design to a fast compiled C++ format along with C++ models of any peripherals.
+
+Extra C++ model files for the simulation, and any custom build steps required, can be added to the ``sim/doit_build.py`` doit build script inside the user project.
+
+A default simulation driver (the C++ code that runs the simulation) is included as `main.cc<example_main_cc>` in the example projects. This code:
```

Comment:
the source reference isn't rendering correctly.  we could fiddle with sphinx.ext.viewcode, but maybe just make a link to github for now?

Reply:
or maybe just make it a literal full path, your call!